lib/ansible/modules/packaging/os/redhat_repository.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (giovannibattistasciortino@gmail.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: redhat_repository
+short_description: Manage RHSM repositories using the C(subscription-manager) command
+description:
+    - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription Management entitlement platform using the C(subscription-manager) command
+version_added: "2.4"
+author: "Giovanni Sciortino (giovannibattistasciortino@gmail.com)"
+notes:
+    - In order to manage rhsm repositories the system must be already registered to rhsm manually or using the ansible module redhat_subscription.
+    - One option between state and list must be defined, both options in the same task must not be defined
+    - When the option state is defined, the option name must be defined
+
+
+requirements:
+    - subscription-manager
+options:
+    state:
+        description:
+          state indicates the desired repositories states.
+        choices: [ "enabled", "disabled" ]
+        required: False
+        default: None
+    name:
+        description:
+          The id of repositories to enable. To operate on several repositories this can accept a comma separated list or a list.
+        required: False
+        default: None
+    list:
+        description:
+          List all/enabled/disabled repository.
+        choices: [ "all", "enabled", "disabled" ]
+        required: False
+        default: None
+
+'''
+
+EXAMPLES = '''
+- name: List all RHSM repositories.
+  redhat_repository:
+    list: all
+  register: rhsm_repository
+
+- name: List enabled RHSM repositories.
+  redhat_repository:
+    list: enabled
+  register: enabled_rhsm_repository
+
+- name: Enable a RHSM repository
+  redhat_repository:
+    name: rhel-7-server-rpms
+    state: enabled
+
+- name: Disable all RHSM repositories
+  redhat_repository:
+    name: '*'
+    state: disabled
+
+- name: Enable all repository starting with rhel-6-server
+  redhat_repository:
+    name: 'rhel-6-server*'
+    state: enabled
+
+- name: Disable all repositories except rhel-7-server-rpms
+  redhat_repository:
+    name: "{{ item }}"
+    state: disabled
+  with_items: "{{ rhsm_repository.repositories | map(attribute='id') | difference(['rhel-7-server-rpms']) }}"
+
+'''
+
+RETURN = '''
+repositories:
+    description: if state is equal to list, rhsm_repositories is the list of repositories
```

Comment:
Indentation issue like in the `DOCUMENTATION` block. Missing capital letter at the beginning and period at the end of the sentence.